compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/ResolveUtils.kt

```diff
@@ -349,3 +353,48 @@ fun BodyResolveComponents.initialTypeOfCandidate(candidate: Candidate): ConeKotl
 private fun initialTypeOfCandidate(candidate: Candidate, typeRef: FirResolvedTypeRef): ConeKotlinType {
     return candidate.substitutor.substituteOrSelf(typeRef.type)
 }
+
+inline val FirCallableDeclaration<*>.containingClass: FirRegularClass?
+    get() = this.containingClassAttr?.let { lookupTag ->
+        session.symbolProvider.getSymbolByLookupTag(lookupTag)?.fir as? FirRegularClass
+    }
+
+val FirFunction<*>.asBadForNamedArgumentTarget: BadNamedArgumentsTarget?
+    get() {
+        if (this is FirConstructor && this.isPrimary) {
+            this.containingClass?.let { containingClass ->
+                if (containingClass.classKind == ClassKind.ANNOTATION_CLASS) {
+                    // Java annotation classes allow (actually require) named parameters.
+                    return null
+                }
+            }
+        }
+        return when (origin) {
+            FirDeclarationOrigin.Source, FirDeclarationOrigin.Library, FirDeclarationOrigin.BuiltIns -> null
```

Comment:
Also, according to https://github.com/JetBrains/kotlin/blob/master/compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/ValueArgumentsToParametersMapper.java#L193 we can return `INTEROP_FUNCTION` for deserialized function whose origin in FIR would be `Library`: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/fir-deserialization/src/org/jetbrains/kotlin/fir/deserialization/FirMemberDeserializer.kt#L353

Reply:
I tried with `ktor` library in a test project and the resolved FirFunction has `Library` as the `origin`. I guess FIR serialization is not the equivalent of the descriptor serialization?

Reply:
Duh... we should revisit this when MPP is supported in FIR. Commenting out `INTEROP_FUNCTION` should be good enough for now. Thanks for the trial and clarification!